### ISSUE TITLE
pi_examples: Building label_image shows missing graph.pb.h then graph.pb.h version error

### ISSUE THREAD

**Issue by avantol** [tensorflow/tensorflow#7905]
When building the label_image example in pi_examples:

./tensorflow/core/framework/graph.pb.h:12:2: error: #error This file was generated by a newer version of protoc which is
./tensorflow/core/framework/graph.pb.h:13:2: error: #error incompatible with your Protocol Buffer headers. Please update

### What related GitHub issues or StackOverflow threads have you found by searching the web for your problem?

None, the mentions of graph.pb.h were all for that file simply being missing. 

In my case, moving graph.pb.h to the location the compiler expects resulted in the version error message, above.

Also, the issue reports seemed to be pre- TF v.1.0.0, and likely not relevant.

### Environment info
Operating System:
Raspbian Jessie, version "Jan 2017", release date "2017-01-11"
https://www.raspberrypi.org/downloads/raspbian/

Installed version of CUDA and cuDNN: 
(please attach the output of `ls -l /path/to/cuda/lib/libcud*`):
N/A

If installed from binary pip package, provide:
N/A

1. A link to the pip package you installed:
2. The output from `python -c "import tensorflow; print(tensorflow.__version__)"`.

If installed from source, provide 

1. The commit hash (`git rev-parse HEAD`)
321d38ee0ff76d749a8578f37128b02bf033ce76

2. The output of `bazel version`
Build label: 0.4.3- (@non-git)
Build target: bazel-out/local-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Fri Feb 24 03:14:34 2017 (1487906074)
Build timestamp: 1487906074
Build timestamp as int: 1487906074

### If possible, provide a minimal reproducible example (We usually don't have time to read hundreds of lines of your code)

Build tensorflow from source, exactly as at:
https://github.com/samjabrahams/tensorflow-on-raspberry-pi/blob/master/GUIDE.md

Attempt tp build label_image example, exactly as at:
https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/pi_examples/

Corrected error caused by missing graph.pb.h by copying graph.pb.h from:
/home/pi/tensorflow/bazel-tensorflow/tensorflow/core/framework/graph.pb.h
to
/home/pi/tensorflow/tensorflow/core/framework/node_def.pb.h

and ran the label_image make again. Error shown below.

### What other attempted solutions have you tried?
None, not very familiar w/Bazel and TF building yet... after all, the pi_examples are for newbies to get started with and learn TF, right? 

### Logs or other output that would be helpful
(If logs are large, please upload as attachment or provide link).

pi@raspberrypi:~/tensorflow $ make -f tensorflow/contrib/pi_examples/label_image/Makefile
gcc --std=c++11 -O0 -I/usr/local/include -I. -I/home/pi/tensorflow/tensorflow/contrib/pi_examples/label_image/../../makefile/downloads -I/home/pi/tensorflow/tensorflow/contrib/pi_examples/label_image/../../makefile/downloads/eigen/ -I/home/pi/tensorflow/tensorflow/contrib/pi_examples/label_image/../../makefile/gen/proto/ -I/home/pi/tensorflow/tensorflow/contrib/pi_examples/label_image/../../makefile/gen/proto_text/ -c tensorflow/contrib/pi_examples/label_image/label_image.cc -o /home/pi/tensorflow/tensorflow/contrib/pi_examples/label_image/gen/obj/tensorflow/contrib/pi_examples/label_image/label_image.o
In file included from tensorflow/contrib/pi_examples/label_image/label_image.cc:32:0:
./tensorflow/core/framework/graph.pb.h:12:2: error: #error This file was generated by a newer version of protoc which is
 #error This file was generated by a newer version of protoc which is
  ^
./tensorflow/core/framework/graph.pb.h:13:2: error: #error incompatible with your Protocol Buffer headers. Please update
 #error incompatible with your Protocol Buffer headers.  Please update
  ^
./tensorflow/core/framework/graph.pb.h:14:2: error: #error your headers.
 #error your headers.
  ^
In file included from tensorflow/contrib/pi_examples/label_image/label_image.cc:32:0:
./tensorflow/core/framework/graph.pb.h:31:51: fatal error: tensorflow/core/framework/node_def.pb.h: No such file or directory
 #include "tensorflow/core/framework/node_def.pb.h"
                                                   ^
compilation terminated.
tensorflow/contrib/pi_examples/label_image/Makefile:79: recipe for target '/home/pi/tensorflow/tensorflow/contrib/pi_examples/label_image/gen/obj/tensorflow/contrib/pi_examples/label_image/label_image.o' failed
make: *** [/home/pi/tensorflow/tensorflow/contrib/pi_examples/label_image/gen/obj/tensorflow/contrib/pi_examples/label_image/label_image.o] Error 1

**Comment by avantol** [comment]
Note: The original problem is that these files were reported by the compiler as missing from 
./tensorflow/core/framework

graph.pb.h
node_def.pb.h
attr_value.pb.h
tensor.pb.h
resource_handle.pb.h
tensor_shape.pb.h
types.pb.h
function.pb.h
op_def.pb.h
versions.pb.h

I read that these files are generated (and then placed in tensorflow/core/framework, i assume), but apparently that didn't happen while building TF.

Rebuilding TF showed that all files were current. 

Running configure (to clean) and make will take over 10 hours (and recovering from numerous Pi "hangs"!), so I haven't done that again yet. 

However, I did get a good build of the TF 1.0.0 Python .whl file, it installed OK, and reports version OK in Python. So, the TF (and Bazel?) build appears to be partially successful.

**Comment by tatatodd** [comment]
@avantol Have you tried following the instructions here to build TF on your pi:
https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/pi_examples
https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/makefile#raspberry-pi

Note that *.pb.h files are generated by the protocol buffer compiler.  That's where to start looking wrt problems with your install.

**Comment by avantol** [comment]
@tatatodd: THANKS!!! Using make (instead of Bazel) to build tensorflow:

https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/makefile#raspberry-pi

worked....  and label_image builds correctly now.

I also had to update protobuf to 3.2.0, others should take note.

My conclusion is that Bazel on Raspberry Pi is a *huge waste of time*, it just doesn't work, likely a problem with defining the build for Raspberry Pi. Also, it crashed Pi frequently (yes, I used the memory tips).  

"make" is 8 hours faster and works correctly.

Also, others should note that the statement
"The recommended way to build TensorFlow from source is using the Bazel open-source build system" at:

https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/makefile#raspberry-pi

does NOT necessarily apply to Raspberry Pi!

This advice took me down completely the wrong road for a week. Beware!
